extensions/wikia/AdEngine/ResourceLoaders/ResourceLoaderAdEngineBlockthroughCode.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+class ResourceLoaderAdEngineBlockthroughCode extends ResourceLoaderAdEngineBase {
+	// 24h: cache for HTTP downloaded code
+	const TTL_SCRIPTS = WikiaResponse::CACHE_STANDARD;
+	// 10m: cache for old file-loaded code
+	const TTL_GRACE = 600;
+	// cache version: increase on any local file change
+	const CACHE_BUSTER = 50;
+	const REQUEST_TIMEOUT = 30;
+
+	const REMOTE_FILE_URL = 'https://wikia-inc-com.videoplayerhub.com/galleryloader.js';
+	const LOCAL_FILE_PATH = __DIR__ . '/../resources/recovery/blockthrough.js';
+
+	protected function getMemcKey() {
+		return wfSharedMemcKey( 'adengine', __METHOD__, static::CACHE_BUSTER );
```

Comment:
Is it unique enough? Isn't it going to conflict with other adengine memc keys?

Reply:
It's gonna return ResourceLoaderAdEngineBlockthroughCode::getMemcKey and key inside this class is used only once, so no, it's not a problem.